app/Http/Api/Controllers/HandleGitHubPullRequestWebhookController.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace App\Http\Api\Controllers;
+
+use App\Domain\Experience\Commands\RegisterPullRequest;
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Validation\UnauthorizedException;
+use Spatie\EventSourcing\Commands\CommandBus;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+
+class HandleGitHubPullRequestWebhookController
+{
+    public function __construct(private CommandBus $bus)
+    {
+    }
+
+    public function __invoke(Request $request)
+    {
+        Log::info("PR webhook received");
+
+        $this->ensureValidRequest($request);
+
+        Log::info("PR webhook valid");
+
+        $payload = json_decode($request->getContent(), true);
+
+        if (! ($payload['pull_request']['merged'] ?? null)) {
+            return;
+        }
+
+        Log::info("Was merged, resolving user…");
+
+        $user = User::whereGithubId($payload['sender']['id'])->first();
+
+        if (! $user) {
+            Log::info("User not found for GH id {$payload['sender']['id']}");
+            return;
+        }
+
+        Log::info("User found, dispatching command.");
+
+        $this->bus->dispatch(RegisterPullRequest::forUser($user, $payload['pull_request']['url'] ?? ''));
+    }
+
+    protected function ensureValidRequest(Request $request): void
+    {
+        if (app()->environment('local')) {
```

Comment:
Thought on using `app()->isLocal()`?

Reply:
Ah, TIL, thanks!

Reply:
Is that some kind of magic method? PhpStorm doesn't know about it…

Reply:
Yeah it's magic ✨

Reply:
In that case I'll stick with what my IDE knows :(

Reply:
```
Illuminate\Foundation\Application
```
Method located in this class